backend/src/main/java/com/redhat/cloud/notifications/processors/email/RecipientResolver.java

```diff
@@ -0,0 +1,34 @@
+package com.redhat.cloud.notifications.processors.email;
+
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.recipients.User;
+import com.redhat.cloud.notifications.recipients.rbac.RbacRecipientUsersProvider;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class RecipientResolver {
+
+    @Inject
+    RbacRecipientUsersProvider rbacRecipientUsersProvider;
+
+    public Uni<List<User>> recipientUsers(String accountId, List<Endpoint> endpoints, Set<String> subscribers) {
+        return Multi.createFrom().iterable(endpoints)
+                .onItem().transformToUni(e -> recipientUsers(accountId, e, subscribers))
+                .concatenate().collect().in(ArrayList<User>::new, List::addAll)
+                .onItem().transform(users -> users.stream().distinct().collect(Collectors.toList()));
+    }
+
+    private Uni<List<User>> recipientUsers(String accountId, Endpoint endpoint, Set<String> subscribers) {
+        Uni<List<User>> usersUni = rbacRecipientUsersProvider.getUsers(accountId, false);
+        return usersUni.onItem().transform(users -> users.stream().filter(user -> subscribers.contains(user.getUsername())).collect(Collectors.toList()));
```

Comment:
Is there a chance `contains` would fail because of the case-sensitive comparison?

Reply:
We take the username from rbac data all the times, it should be the same case. I'm not sure if different cases means different users, might need to check that.

Reply:
I tried login with my username in an alternate case and it worked.
Changed to use anyMatch -> equalsIgnoreCase to be on the safe side